### PR TITLE
User Story 3 . Browsing Items by category

### PR DESCRIPTION
**Pull request by tjp2021**
Fixes/Addresses: #4 

Changes made: 
 - Created Categories Controller with Show method
 - Added custom routes to routes.rb
 -Built basic show page to test custom routes.
- Created Category model

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,5 +5,7 @@
 
   namespace :admin do
     resources :items, only: [:index, :new, :create]
+
+  get '/:name' => 'categories#show', as: :seecategory
```
Comment (adrian-lara) [inline config/routes.rb]: @Tyjoo26 we can move this outside of the admin namespace

**Review by adrian-lara — CHANGES_REQUESTED** [review]
We should have some feature tests in place.
Also a reminder to rename your user model test to have '_spec' at the end.

**Comment by tjp2021** [comment]
Reviewed your comments Adrian, I've moved the route to where you recommended, and created a feature test for User Story #3. I'm going to make the test more robust

**Comment by tjp2021** [comment]
As per Adrians recommendation here are the additional modifications with this revised Pull Request

-Made new feature test for items associated with specific category being displayed on categories page that is hidden by vanity url
-Created new migration for add reference from Category to item
-Modified Model files for both Category and Item to account for new relationship
-Changed categories controller show method to account for relationship between category and item. Also added a downcase method to make the vanity url more flexible in handling dynamic client input.

**Review by adrian-lara — APPROVED** [review]

**Pull request merged**